src/RetryAnnotationTrait.php

```diff
@@ -23,46 +26,39 @@ private function getRetryAttemptsAnnotation()
             $retries = $annotations['class']['retryAttempts'][0];
         }
 
-        return $this->validateRetryAttemptsAnnotation($retries);
+        return $this->parseRetryAttemptsAnnotation($retries);
     }
 
-    /**
-     * @return int
-     * @throws LogicException
-     */
-    private function validateRetryAttemptsAnnotation($retries)
+    private function parseRetryAttemptsAnnotation(string $retries): int
     {
         if ('' === $retries) {
-            throw new LogicException(
+            throw new InvalidArgumentException(
                 'The @retryAttempts annotation requires an integer as an argument'
             );
         }
         if (false === is_numeric($retries)) {
-            throw new LogicException(sprintf(
+            throw new InvalidArgumentException(sprintf(
                 'The @retryAttempts annotation must be an integer but got "%s"',
                 var_export($retries, true)
             ));
         }
-        if (floatval($retries) != intval($retries)) {
-            throw new LogicException(sprintf(
+        if ((float) $retries !== (float) (int) $retries) {
```

Comment:
Mostly CS at this point, but casting is more common rather than relying on `floatval()` and the likes; casting is also slightly faster